src/solvers/base_solver.cpp

```diff
@@ -25,98 +25,124 @@ BaseSolver::BaseSolver(MPI_Comm comm, int n) : BaseSolver(comm)
 }
 
 void BaseSolver::SetEssentialBCs(const std::vector<int> &                 ess_bdr,
-                                 std::shared_ptr<mfem::VectorCoefficient> ess_bdr_vec_coef, int component)
+                                 std::shared_ptr<mfem::VectorCoefficient> ess_bdr_vec_coef, 
+                                 mfem::ParFiniteElementSpace &fes, 
+                                 int component)
 {
-  auto bc_data = std::make_shared<BoundaryConditionData>();
+  auto bc = std::make_shared<BoundaryCondition>();
 
-  bc_data->bc_markers.SetSize(ess_bdr.size());
+  bc->markers.SetSize(ess_bdr.size());
 
   for (unsigned int i = 0; i < ess_bdr.size(); ++i) {
-    bc_data->bc_markers[i] = ess_bdr[i];
+    bc->markers[i] = ess_bdr[i];
+
+   if (bc->markers[i] == 1) {
+      for (auto &existing_bc : m_ess_bdr) {
+        if (existing_bc->markers[i] == 1) {
+          mfem::mfem_warning("Multiple definition of essential boundary! Using first definition given.");
+          bc->markers[i] = 0;
+          break;
+        }
+      }
+    }    
   }
 
-  bc_data->vec_coef  = ess_bdr_vec_coef;
-  bc_data->component = component;
+  bc->vec_coef  = ess_bdr_vec_coef;
+  bc->component = component;
 
-  m_ess_bdr.push_back(bc_data);
+  fes.GetEssentialTrueDofs(bc->markers, bc->true_dofs, component);
+
+  m_ess_bdr.push_back(bc);
 }
 
 void BaseSolver::SetTrueDofs(const mfem::Array<int> &                 true_dofs,
                              std::shared_ptr<mfem::VectorCoefficient> ess_bdr_vec_coef)
 {
-  auto bc_data = std::make_shared<BoundaryConditionData>();
+  auto bc = std::make_shared<BoundaryCondition>();
 
-  bc_data->bc_markers.SetSize(0);
+  bc->markers.SetSize(0);
```

Comment:
Just curious... why is this calling SetSize(0)? Seems that is what the default constructor should be doing... This is quite the nitpick btw.

Reply:
In theory yes, but I've been burned by the memory management of the `mfem::Array` class before.